java-source/src/main/java/net/corda/examples/obligation/ObligationContract.java

```diff
@@ -107,7 +105,7 @@ private void verifySettle(LedgerTransaction tx, Set<PublicKey> signers) {
             // Sum the cash being sent to us (we don't care about the issuer).
             Amount<Currency> sumAcceptableCash = withoutIssuer(sumCash(acceptableCash));
             Amount<Currency> amountOutstanding = inputObligation.getAmount().minus(inputObligation.getPaid());
-            req.using("The amount settled cannot be more than the amount outstanding.", amountOutstanding.getQuantity() >= sumAcceptableCash.getQuantity());
+            req.using("The amount settled cannot be more than the amount outstanding.", amountOutstanding.compareTo(sumAcceptableCash) >= 0);
```

Comment:
The tests for multiple currencies were failing. In the kotlin version, the >= invokes the Amount.compareTo function, which runs the checkToken function, which checks that the two currencies are the same. I don't think this is particularly clear to read, but it's the only way of doing it via the existing api. the checkToken function on Amount is private.